modules/lsp/servers/default.nix

```diff
@@ -191,5 +191,33 @@ in
 
       # Propagate per-server warnings
       warnings = lib.mkIf (serverWarnings != [ ]) serverWarnings;
+
+      assertions = [
+        (
+          # Assert that we don't have any redundant custom modules
+          let
+            serverOptions = lib.pipe options.lsp.servers [
+              (opt: opt.type.getSubOptions opt.loc)
+            ];
+            modules = lib.pipe ./custom [
+              builtins.readDir
+              (lib.filterAttrs (name: _: !(serverOptions ? ${lib.strings.removeSuffix ".nix" name})))
+              (lib.mapAttrsToList (
+                name: type: ./custom/${name} + lib.optionalString (type == "directory") "/default.nix"
+              ))
+              (builtins.filter (lib.strings.hasSuffix ".nix"))
+              (builtins.filter lib.pathExists)
+              (map (module: lib.strings.removePrefix (toString ../../../. + "/") (toString module)))
+            ];
+          in
+          {
+            assertion = modules == [ ];
+            message = ''
+              The following custom modules do not correspond to an LSP server option:
+              ${lib.concatMapStringsSep "\n" (module: "- ${module}") modules}
+            '';
+          }
+        )
+      ];
```

Comment:
Maybe this assertion should be a dedicated test, or part of `test-sources/modules/lsp.nix`?

This should never trigger on user's nixvim configurations, so it's a waste of their resources to evaluate the assertion.